pkg/services/publicdashboards/queries/dashboard_queries_test.go

```diff
@@ -565,3 +565,19 @@ func TestSanitizeMetadataFromQueryData(t *testing.T) {
 		}
 	})
 }
+
+func groupQueriesByDataSource(t *testing.T, queries []*simplejson.Json) (result [][]*simplejson.Json) {
+	t.Helper()
+	byDataSource := make(map[string][]*simplejson.Json)
+
+	for _, query := range queries {
+		uid := GetDataSourceUidFromJson(query)
```

Comment:
@ryantxu did data sources always have UIDs? just wondering what happens if older dashboards are imported or if a user upgrades with an older saved dashboard in their DB.

Reply:
I checked out queries from Grafana 7 and it looks like instead of doing `query.datasource.uid`, it sends `query.datasourceId`, which the backend code looks for (not something I did, was already in there).